CMakeLists.txt

```diff
@@ -34,7 +34,15 @@ option(WERROR "Treat compiler warnings as errors" OFF)
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake/Modules")
 
-option(BUILD_IDLC "Build IDL preprocessor" ON)
+if (CMAKE_CROSSCOMPILING)
+  option(BUILD_IDLC "Build IDL preprocessor (default:on, default-crosscompiling:off)" OFF)
```

Comment:
I'm wondering if this would work: `option(BUILD_XXX "description" ${CMAKE_CROSSCOMPILING})`. Not a fan of stating the option itself twice...